account/identity.go

```diff
@@ -140,10 +142,18 @@ func (m *GormIdentityRepository) Create(ctx context.Context, model *Identity) er
 	}
 	err := m.db.Create(model).Error
 	if err != nil {
-		goa.LogError(ctx, "error adding Identity", "error", err.Error())
+		log.LogError(ctx, map[string]interface{}{
+			"identity": model,
+			"err":      err.Error(),
+		}, "Unable to create the identity")
 		return errors.WithStack(err)
```

Comment:
I'm saying that it makes no sense to log a `not found error` for a work item for instance, when somebody searches for a particular work item. The error `NotFound` is handled at the caller level and will be returned as an empty list. This only works for application specific errors I think. Genuine `error` errors need to be inspected for their actual type.

We should enhance each errror with a context if we return it to the caller.

Let's have a discussion with @aslakknutsen on this.